### DISCUSSION TITLE
Compile for windows using Cygwin

### DISCUSSION THREAD

**Discussion by BrunoRNS** [vhelin/wla-dx#632] [Ideas]
With cmake and make installed in Cygwin you can use `cmake .` and after `make`. It will automaticly compile into a .exe files the wla-dx. This can free us of installing Visual Studio and open the .sln solution. I think it would be nice to be in INSTALL guide, new users can give up for installing because they can think it only works with Visual Studio. I tried to use the web installer but didn't work, so, for me with Cygwin in windows was the easily way.

**Comment by vhelin** [comment]
Just rechecked INSTALL and it says how to use cmake for e.g., Cygwin plus also that there is a Visual Studio solution... ?

**Comment by vhelin** [comment]
Also: what is that web installer you tried?

**Comment by BrunoRNS** [comment]
in the installation guide it says to use the -G option indicating the target platform from what I understand. But if we indicate the target platform to be UNIX-LIKE( like installation guide indicate ), it will compile to unix-like, but if we don't indicate anything, it will try to create an executable for windows, if it is on windows. That's all I thought if we put it in it would be clearer.

Replying from the installer, I think I got confused, sorry, I thought that when there were "x64 releases" and "x86 releases" for windows in README.md, I thought it was the binary ready to run, but it was my mistake.

Instead, we can talk about using cmake for any system: something like this: "Cmake will automatically compile to your system's architecture without passing the target platform argument: `cmake .`"

Overall, I think the installation guide looks great. But for beginners, like me who even confused if it was an web installer (the one from Azure), I thought there could be some simpler way to pass the executable, or give this option to use cmake without indicating the target platform.

**Comment by BrunoRNS** [comment]
`First make sure you have gcc, g++, make and cmake installed. To generate
the makefiles, go to the wla-dx main folder, and write

  cmake -G "Unix Makefiles" .  ; under Unix / OS X / Cygwin / MSYS2
  cmake -G "MSYS Makefiles" .  ; under MSYS / MinGW
  cmake -G "MinGW Makefiles" . ; under (plain) MinGW
  cmake -G "NMake Makefiles" . ; if you wish to use NMake

After CMake has done its job, just run

  make

You will get all the executables in the "binaries" folder. Or with

  make install

all the executables will be copied to /usr/local/bin

PS. If you are running Windows, and want to create Windows executables,
you can find "WLA DX.sln" Visual Studio solution under windows folder.` 

Up here we have the beginning of the installation guide. If we say that cmake generates the Makefile for your system automatically `cmake .` ,the user won't need to focus on the Makefile platform. But that's just on cygwin, I haven't tested it on other systems to see if it does this on its own. I didn't see the cmake documentation, maybe it's recommended to use -G option, if so, I was wrong, but we can try to create a web installer to be be easier.

If you want, I can create the code of an installer on the web here, then you upload the binaries to google drive and I put them on a free server like github pages one (https://docs.github.com/pt/pages/getting-started-with-github-pages/about-github-pages). After I send you the link of the server and you can add to README.md.  

Please tell me if you want the web installer, if yes I can do it soon.

**Comment by BrunoRNS** [comment]
I will close this ideia discussion because isn't my ideia isn't helpful for that, sorry guy for this. I'm currently working in a SNES-IDE, when I finish, I hope you see that.

**Comment by vhelin** [comment]
Sorry, I'll look into this later this week - just so busy with other things at the moment...
